controllers/openshift_controller.go

```diff
@@ -1946,6 +1948,22 @@ func (r *KataConfigOpenShiftReconciler) resetInProgressCondition() {
 	r.Log.Info("InProgress Condition reset")
 }
 
+func (r *KataConfigOpenShiftReconciler) isInstalling() bool {
+	cond := r.findInProgressCondition()
+	if cond == nil {
+		return false;
+	}
+	return cond.Status == corev1.ConditionTrue && cond.Reason == "Installing";
+}
+
+func (r *KataConfigOpenShiftReconciler) isUpdating() bool {
+	cond := r.findInProgressCondition()
+	if cond == nil {
+		return false;
+	}
+	return cond.Status == corev1.ConditionTrue && cond.Reason == "Updating";
+}
+
```

Comment:
I'm wondering if go is smart enough to detect that the code for these is identical.

Reply:
I'm not quite sure if I understand you correctly but `isInstalling()` checks for `cond.Reason == "Installing"` whereas `isUpdating()` checks for `cond.Reason == "Updating"`, doesn't it?

Reply:
Yes, same code with _just_ a different literal string. Wondering if go can detect that and factor out the code internally.

Reply:
I see. I've no idea but honestly, I doubt it. :-)